### ISSUE TITLE
Protect: Link fails in email sent to unlock site

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#8322]
Related to #8117 

I had two reports of this this morning, in 831912-zen and 831859-zen.

In both cases, the users got the email to let them log in to their site. Upon clicking the login link, they got the following error:

`{'error':'Bad Request','message':'Invalid input.'}`

Note that 1 ticket was of a user that wasn't trusting REMOTE_ADDR, and the other was someone with many failed logins from their IP so they were blocked.

**Comment by jeherve** [comment]
cc @enejb

**Comment by enejb** [comment]
The user see `{'error':'Bad Request','message':'Invalid input.'}` message after the login link expires of if they type the wrong link. Retrying again would work out for them. 

cc: @lezama Should we increase the duration of that token?

**Comment by richardmtl** [comment]
How long is the duration of the token, @enejb ? I don't know how long these folks waited; I would guess not a long time but I don't know for a fact.

**Comment by enejb** [comment]
The link is only valid for 5minutes. 

It also says so in the email that we send out to the user.
```
Clicking the link below will allow you to either attempt additional login attempts or change your password for your WordPress site for 5 minutes from the ip address %2$s.
```
As I understand things users are usually only blocked for 20 minutes. So having something that is super long is probably not very useful. 

But in the case where the token is not valid any more that doesn't seem to be the case.

**Comment by richardmtl** [comment]
<blockquote>As I understand things users are usually only blocked for 20 minutes. So having something that is super long is probably not very useful.

</blockquote>


Actually, no. Users can be blocked for much longer, ESPECIALLY in cases where we are trusting the wrong header (which is the majority of cases when users get blocked); in those cases, blocks can be effectively indefinite (the block expires and then are blocked again almost immediately, and for longer). At least, that's how *I* understand it, so..  I think 5 minutes is not long enough. Especially when a user is frazzled and locked out and not sure what's going on, they might not go check their email right away.

**Comment by enejb** [comment]
I updated the email link duration to 15 minutes. 

Can you explain a bit more about the wrong header? What is the issue there?

**Comment by jeherve** [comment]
> Can you explain a bit more about the wrong header? What is the issue there?

This should explain things a bit: PCYsg-8f0-p2

You can also look at 1257-jpop-issues for some recent examples of this issue.

**Comment by enejb** [comment]
Thanks for the links. 
It would be good to implement some sort of solution where we automatically set the IP address in a better way. I know it really depends on the configuration of the server. 

But maybe we can automate this somehow. For example if we know the IP address of the servers and make a request to the jetpack server. We could use the known predefined array of possible keys and headers and then try to configure the IP address that matched the one that we expect. 

cc: @samhotchkiss What do you think?

**Comment by merire111** [comment]
Same problem here, unlock site link not working. I'm clicking within 5 minutes but still. I have to login to wordpress.com, go to settings, security, and disable protection, before logging in and activating protection again. 
Adding my IP adress to the whitelist does not solve the issue.

**Comment by richardmtl** [comment]
@merire111 Your site likely needs us to tweak a setting on our end. Can you contact us directly please? 

https://jetpack.com/contact-support/

**Comment by Jzatopa** [comment]
Same problem here too, unlock site link doesn't work.  Not sure why this would be opt out and not opt in.

**Comment by StefMattana** [comment]
Same issue on 900393-zen
In here, the user wasn't trusting REMOTE_ADDR so I switched to it.

**Comment by StefMattana** [comment]
Also reported in 899797-zen
The user wasn't trusting REMOTE_ADDR so I switched to it.

**Comment by richardmtl** [comment]
@enejb It is my understanding that if the Headers that we are looking at are incorrect on our end, then the link won't work. I don't know what percentage of people who get blocked are blocked because they mistakenly enter the wrong password (and thus the link is useful for them), versus the people blocked  because the headers we are trusting are wrong (and thus the link won't work for them). However, right now we are giving all the people with incorrect trusted headers "false hope" that they can get into their site by clicking the link we send them. I suppose there's no easy way to filter those out so we don't even send them a link and we just say "contact us please"?

**Comment by enejb** [comment]
I think there might be a way to filter them out. Will look into this some more. Great suggestion!

**Comment by enejb** [comment]
Also I think this should be fixed via the Jetpack UI somehow. Telling the user something is wrong and what they should do to fix it. (Contact support)

**Comment by enejb** [comment]
Also I think this should be fixed via the Jetpack UI somehow. Telling the user something is wrong and what they should do to fix it. (Contact support) 
cc: @samhotchkiss and @MichaelArestad :)

**Comment by jeherve** [comment]
More discussion at p7rcWF-HJ-p2

**Comment by vrocks** [comment]
I got this request when trying to login. Had it sent to email. Got this:
{"error":"Bad Request","message":"Invalid input."}

And I did it all in minutes. So no timeout on the request. I'd say your plugin is broken.

**Comment by jeherve** [comment]
@vrocks Could you contact us via [this contact form](https://jetpack.com/contact-support/?rel=support) and give us more details about your site? We'll help you get this fixed on your site.

Thanks!

**Comment by joendotcom** [comment]
979688-zen

**Comment by tmmbecker** [comment]
1065607-zen

**Comment by chromechris** [comment]
As someone else stated, I also followed the link probably 2 minutes or less after receiving the link, and got the same error API response. This looks like a plugin issue.

**Comment by chromechris** [comment]
@jeherve I have contacted support through the provided link. I hope there is a fix for this :( Thanks for the help.

**Comment by 42Rincewind** [comment]
I encountered this issue today. I bypassed it by disabling the Jetpack plugin manually using phpMyAdmin. I then could log into my site. I reenabled Jetpack in my site and turned off the "Brute force attack protection" feature. I have other security plugins installed so it was superfluous. I live in China and often use VPNs to access the web, changing IP regularly, so whitelisting all my IP addresses isn't practical.

**Comment by joendotcom** [comment]
1152270-zen

**Comment by richardmtl** [comment]
@42Rincewind In your case, I suspect the VPN IPs were the ones getting blocked, but we could always verify it on our end if you send us your site's URL (and ideally, some or all of the IPs that were blocked) here: https://jetpack.com/contact-support/  Of course, since you turned off Protect, it's up to you if you want us to follow up on this! :)

**Comment by chaselivingston** [comment]
Also reported in 1162780-zen

**Comment by tmmbecker** [comment]
221045-zen

**Comment by tmmbecker** [comment]
1232562-zen

**Comment by chaselivingston** [comment]
1238231-zen

**Comment by tmmbecker** [comment]
1254295-zen

**Comment by ntpixels** [comment]
1261281-zen

**Comment by jesscga** [comment]
1269716-zen

**Comment by jesscga** [comment]
1286665-zen

**Comment by tmmbecker** [comment]
1289550-zen

**Comment by StefMattana** [comment]
1302552-zen

**Comment by StefMattana** [comment]
1312971-zen

**Comment by StefMattana** [comment]
1319911-zen

**Comment by chaselivingston** [comment]
1319950-zen

**Comment by StefMattana** [comment]
1322772-zen

**Comment by chaselivingston** [comment]
1325354-zen

**Comment by StefMattana** [comment]
1308737-zen

**Comment by danjjohnson** [comment]
1330881-zen

**Comment by StefMattana** [comment]
Also reported on https://wordpress.org/support/topic/jetpack-blocking-websites-ip/

**Comment by StefMattana** [comment]
1333615-zen

**Comment by tmmbecker** [comment]
1341560-zen

**Comment by tmmbecker** [comment]
1349711-zen

**Comment by amy6147** [comment]
I have been experiencing this same issue. My IP in whatsmyip is showing another IP wherein the Jetpack has blocked another IP and blocked the access.
 
an email was sent to unblock to the verified user. 

When I clicked on the link provided to login - it gave the same error 
{'error':'Bad Request','message':'Invalid input.'}

It would be not good if I want the users to subscribe on the website and they will be blocked to login just because of this. 

Can't we do something about that.

**Comment by abidhahmed** [comment]
Hey @amy6147 ~ could you contact us through [this form](https://jetpack.com/contact-support/?rel=support)? Please include info about your sites, as well as the IP addresses that are blocked. We'll help take care of this for you. Do make sure to include more info, as your comment states that different IPs were blocked.

Thanks!

**Comment by coder-karen** [comment]
1381584-zen

**Comment by joendotcom** [comment]
1381855-zen

**Comment by StefMattana** [comment]
1407858-zen

**Comment by htdat** [comment]
Note: This issue has the detailed research on our private discussion p9dueE-mO-p2

**Comment by htdat** [comment]
1410269-zen

**Comment by jeherve** [comment]
Changing the priority on this one; while it is an annoying problem for someone getting blocked out of their site, it seems the issue only happens with mis-configured servers that report the wrong IP address in headers like `HTTP_X_FORWARDED_FOR`. We can fix it on our end by looking at the list of headers given by the server and only looking at the one that seems to report the right address.

If you run into issues like this on your own site, please contact support via [this contact form](https://jetpack.com/contact-support/?rel=support) and let us know your site URL; we will fix things for you.

I'll leave this issue opened as we may be able to circumvent some of those issues in the future by improving our IP detection process.

**Comment by htdat** [comment]
1442928-zen

**Comment by tmmbecker** [comment]
1492300-zen

**Comment by joendotcom** [comment]
1492369-zen

**Comment by StefMattana** [comment]
1512522-zen

**Comment by supernovia** [comment]
https://wordpress.org/support/topic/admin-access-blocked-by-jetpack/

**Comment by supernovia** [comment]
https://wordpress.org/support/topic/jetpack-keeps-locking-login-page-broken-link-in-help-email/

**Comment by joendotcom** [comment]
1533692-zen

**Comment by jeremypaavola** [comment]
1556889-zen

**Comment by brucea8c** [comment]
1567653-zen

**Comment by StefMattana** [comment]
1578156-zen

**Comment by htdat** [comment]
1583112-zen

**Comment by StefMattana** [comment]
1583373-zen

**Comment by joendotcom** [comment]
1584262-zen

**Comment by StefMattana** [comment]
1585183-zen

**Comment by jeremypaavola** [comment]
1594387-zen

**Comment by jeremypaavola** [comment]
1581800-zen

**Comment by jeremypaavola** [comment]
1592368-zen

**Comment by StefMattana** [comment]
1596242-zen

**Comment by StefMattana** [comment]
1593064-zen

**Comment by chaselivingston** [comment]
1598045-zen

**Comment by mzakariya** [comment]
1597607-zen (not Bluehost, but unclear who the host is; seems like a reseller)

**Comment by lizkarkoski** [comment]
There is another report of this error. 
https://wordpress.org/support/topic/jetpack-has-locked-your-sites-login-page-6/#post-10975357

**Comment by josephrocca** [comment]
I got this today when I tried to use cloudflare in front of my server, so I guess it's something to do with header stuff. I clicked the link within 30 seconds or so, so it's definitely not related to link expiry.

**Comment by ntpixels** [comment]
1715467-zen

**Comment by codestor4** [comment]
1792502-zen

**Comment by Prizem** [comment]
I also have this issue on one of my sites that I still have this protection on. I've had the issue for over a year. Corresponding with Jetpack support, they just pointed me to this thread to track progress. Looks like not much progress has been made in a year.

**Comment by richardmtl** [comment]
@Prizem We should be able to fix the underlying issue with your site(s); can you share a ticket number of your correspondence with us, so I can take a look?

**Comment by Prizem** [comment]
> @Prizem We should be able to fix the underlying issue with your site(s); can you share a ticket number of your correspondence with us, so I can take a look?

Sure! Looks like tickets 1792502 and 1657826.

**Comment by StefMattana** [comment]
#1835534-zen Hostis

**Comment by tmmbecker** [comment]
#1846906

**Comment by RCowles** [comment]
1857517-zen

**Comment by StefMattana** [comment]
1865106-zen

**Comment by ntpixels** [comment]
1876781-zen

**Comment by RCowles** [comment]
1880282-zen

**Comment by tmmbecker** [comment]
1889229-zen

**Comment by KokkieH** [comment]
Another in https://wordpress.org/support/topic/locked-out-of-jetpack-3/

**Comment by KokkieH** [comment]
Another in https://en.forums.wordpress.com/topic/jetpack-locked-out/

**Comment by immersiontravis** [comment]
Same problem here (on both the QA and production instances of my site).

**Comment by danjjohnson** [comment]
1895167-zen

**Comment by richardmtl** [comment]
@immersiontravis We can fix it for you if you contact us with your site URLs: https://jetpack.com/contact-support/

**Comment by danjjohnson** [comment]
1897156-zen

**Comment by jeremypaavola** [comment]
1901568-zen

**Comment by StefMattana** [comment]
1906075-zen

**Comment by coder-karen** [comment]
1907329-zen

**Comment by ntpixels** [comment]
1908207-zen

**Comment by brucea8c** [comment]
1910888-zen

**Comment by coder-karen** [comment]
1912884-zen

**Comment by coder-karen** [comment]
1912666-zen

**Comment by RCowles** [comment]
1915822-zen

**Comment by RCowles** [comment]
1916497-zen

**Comment by RCowles** [comment]
1916606-zen

**Comment by coder-karen** [comment]
1917036-zen

**Comment by coder-karen** [comment]
1917292-zen

**Comment by StefMattana** [comment]
1917065-zen

**Comment by danjjohnson** [comment]
1917022-zen

**Comment by coder-karen** [comment]
1917435-zen

**Comment by danjjohnson** [comment]
1917552-zen

**Comment by brucea8c** [comment]
1917897-zen

**Comment by jeremypaavola** [comment]
1919171-zen

**Comment by jeremypaavola** [comment]
1920016-zen

**Comment by jeremypaavola** [comment]
1920148-zen

**Comment by jeremypaavola** [comment]
1921777-zen

**Comment by coder-karen** [comment]
1913219-zen

**Comment by coder-karen** [comment]
1930569-zen

**Comment by StefMattana** [comment]
1930890-zen

**Comment by coder-karen** [comment]
1930846-zen

**Comment by tmmbecker** [comment]
1930986-zen

**Comment by StefMattana** [comment]
1917068-zen

**Comment by RCowles** [comment]
1932985-zen

**Comment by RCowles** [comment]
1933178-zen

**Comment by StefMattana** [comment]
1933764-zen

**Comment by coder-karen** [comment]
1934432-zen

**Comment by RCowles** [comment]
1935157-zen

**Comment by StefMattana** [comment]
1936537-zen

**Comment by coder-karen** [comment]
1936915-zen

**Comment by RCowles** [comment]
1940634-zen

**Comment by kraftbj** [comment]
r190045-wpcom improves the e-mail's handling and should resolve some (if not all) of these issues.

**Comment by supernovia** [comment]
Woot woot! Thanks :)

**Comment by jeremypaavola** [comment]
1946446-zen

**Comment by kraftbj** [comment]
From checking with Happiness, we've had no new recent reports so I am marking as resolved.

**Comment by lezama** [comment]
❤️

**Comment by htdat** [comment]
Ticket: 2043791-zen 
Blog ID: 132029627 
Jetpack: 7.3.1
Nameserver: Hostgator

@kraftbj - can you check this case? Also, should I open a new ticket for this or continue here?

**Comment by bikedorkjon** [comment]
Ticket: 20716810-zen @jeherve since Kraft is AFK can you assign someone to own this possibly?

**Comment by jeherve** [comment]
@bikedorkjon This seemed to be a problem with the server configuration; it is now fixed within the Protect API settings for that site.

**Comment by bikedorkjon** [comment]
They responded that things were fine, just forgot to close this.
